config/logstash.yml

```diff
@@ -236,6 +236,7 @@
 #xpack.monitoring.elasticsearch.sniffing: false
 #xpack.monitoring.collection.interval: 10s
 #xpack.monitoring.collection.pipeline.details.enabled: true
+#xpack.monitoring.collection.write_direct.enabled: true
```

Comment:
I generally agree with you @robbavey, I would also change the name from `write_direct` to `direct_shipping`, but I have some doubts to clear up:
- to start monitoring we have the `xpack.monitoring.enabled` settings that has to be enabled to use the internal collectors. So we could potentially have a `monitoring.collection.direct_shipping.enabled` (more specific) that require a more general `xpack.monitoring.enabled` to work. But xpack is an extension pack, so we would have something in `monitoring.*` path the require something form extension pack `xpack.monitoring.*` to work. I would expect that both stay at the same level, what do you think?

Reply:
If you look at what the Beats team did - they have [three options](https://www.elastic.co/guide/en/beats/filebeat/current/monitoring.html) for monitoring, [Legacy](https://www.elastic.co/guide/en/beats/filebeat/current/monitoring-internal-collection-legacy.html) internal pointing to production ES cluster, [regular](https://www.elastic.co/guide/en/beats/filebeat/current/monitoring-internal-collection.html) internal monitoring pointing to monitoring ES cluster and [Metricbeat](https://www.elastic.co/guide/en/beats/filebeat/current/monitoring-metricbeat-collection.html) collection.

How they appear to have structured it, is to have the [Legacy monitoring](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-monitor-legacy.html) being configured by settings in the `xpack.monitoring` namespace, and the ['direct' monitoring](https://www.elastic.co/guide/en/beats/filebeat/current/monitoring-internal-collection.html) being configured by settings in the `monitoring` namespace, rather than having a specific setting for `direct_shipping`. To enable legacy monitoring, `xpack.monitoring.enabled` would be used, and to use direct monitoring, `monitoring.enabled`

Reply:
Nice, deprecating the Legacy they redirect to the new and removed the documentation for the settings redirecting to the new internal collectors for reference. So we could:
- keep the handling of old `xpack.monitoring` settings for back compatibility
- copy the old setting in `xpack.monitoring.*` to the `monitoring.*` where `monitoring.enabled` means "use the ship direct way"
- adapt the documentation to reflect this

Reply:
Yes, I think that makes sense.

Reply:
@robbavey with commit 5084eb1cc54336f4a4562f264d62c747cc12698b I've done the rename, please, let me know what do you think about it

Reply:
I think the rename makes sense - @chrisronline, @ycombinator, does this make sense to you, and is it consistent with the rest of the stack in terms of adding direct to monitoring cluster functionality?

Reply:
Yes, `monitoring.*` settings is exactly what we're pushing for consistency across the stack 👍